src/test/java/api/requests/RequestScheduledNoticesTests.java

```diff
@@ -0,0 +1,342 @@
+      isEquivalentTo(requestExpiration.toDateTimeAtStartOfDay()));
+    assertThat(noticeConfig.getString("timing"), is("Upon At"));
+    assertThat(noticeConfig.getString("templateId"), is(templateId.toString()));
+    assertThat(noticeConfig.getString("format"), is("Email"));
+    assertThat(noticeConfig.getBoolean("sendInRealTime"), is(true));
+
+    requestsClient.replace(request.getId(), requestBuilder.withRequestExpiration(requestExpiration.plusDays(1)));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, hasSize(1));
+
+    scheduledNotices = scheduledNoticesClient.getAll();
+    assertThat(scheduledNotices.size(), is(1));
+
+    scheduledNotice = scheduledNotices.get(0);
+    noticeConfig = scheduledNotice.getJsonObject("noticeConfig");
+
+    assertThat(scheduledNotice.getString("requestId"), is(request.getId().toString()));
+    assertThat(scheduledNotice.getString("triggeringEvent"), is("Request Expiration"));
+    assertThat(scheduledNotice.getString("nextRunTime"),
+      isEquivalentTo(requestExpiration.toDateTimeAtStartOfDay().plusDays(1)));
+    assertThat(noticeConfig.getString("timing"), is("Upon At"));
+    assertThat(noticeConfig.getString("templateId"), is(templateId.toString()));
+    assertThat(noticeConfig.getString("format"), is("Email"));
+    assertThat(noticeConfig.getBoolean("sendInRealTime"), is(true));
+  }
+
+  @Test
+  public void holdShelfExpirationNoticeShouldBeScheduledOnCheckIn()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    JsonObject requestNotice = new NoticeConfigurationBuilder()
+      .withTemplateId(templateId)
+      .withHoldShelfExpirationEvent()
+      .withUponAtTiming()
+      .sendInRealTime(true)
+      .create();
+
+    NoticePolicyBuilder noticePolicyBuilder = new NoticePolicyBuilder()
+      .withName("request policy")
+      .withRequestNotices(Collections.singletonList(requestNotice));
+
+    useLoanPolicyAsFallback(
+      loanPoliciesFixture.canCirculateRolling().getId(),
+      requestPoliciesFixture.pageRequestPolicy().getId(),
+      noticePoliciesFixture.create(noticePolicyBuilder).getId());
+
+    LocalDate requestExpiration = LocalDate.now().plusMonths(3);
+    RequestBuilder requestBuilder = new RequestBuilder().page()
+      .forItem(item)
+      .withRequesterId(requester.getId())
+      .withRequestDate(DateTime.now())
+      .withStatus(OPEN_NOT_YET_FILLED)
+      .withPickupServicePoint(pickupServicePoint)
+      .withRequestExpiration(requestExpiration);
+    IndividualResource request = requestsFixture.place(requestBuilder);
+
+    CheckInByBarcodeRequestBuilder at = new CheckInByBarcodeRequestBuilder()
+      .forItem(item)
+      .withItemBarcode(item.getBarcode())
+      .at(pickupServicePoint);
+
+    loansFixture.checkInByBarcode(at);
+    String holdShelfExpirationDate = requestsClient.get(request)
```

Comment:
I think these two statements could be more readably when separated by a blank line

Reply:
done